src/main/java/org/openrewrite/java/spring/boot2/search/IntegrationSchedulerPoolRecipe.java

```diff
@@ -0,0 +1,185 @@
+                .collect(Collectors.toSet());
+
+        Set<JavaProject> projectsWithCommentOverProperty = new HashSet<>();
+
+        // Leave the comment about scheduler pool size next to 'spring.task.scheduling.pool.size' property in the external properties file (props or yaml)
+        List<SourceFile> modified = ListUtils.map(before, source -> {
+            String fileName = source.getSourcePath().getFileName().toString();
+            JavaProject javaProject = source.getMarkers().findFirst(JavaProject.class).orElse(null);
+            if (javaProjects.contains(javaProject)) {
+                if (APP_PROPS_FILE_REGEX.matcher(fileName).matches() && source instanceof Properties) {
+                    Set<Properties.Entry> foundEntries = FindProperties.find((Properties) source, PROPERTY_KEY, false);
+                    if (!foundEntries.isEmpty()) {
+                        projectsWithCommentOverProperty.add(javaProject);
+                        // There should only be one exact match!
+                        Properties.Entry entry = foundEntries.iterator().next();
+                        return (SourceFile) new PropertiesVisitor<ExecutionContext>() {
+                            @Override
+                            public Properties visitFile(Properties.File file, ExecutionContext context) {
+                                int idx = file.getContent().indexOf(entry);
+                                if (idx >= 0) {
+                                    Properties.Comment comment = new Properties.Comment(Tree.randomId(), "\n", Markers.EMPTY, PROPS_MIGRATION_MESSAGE);
+                                    List<Properties.Content> contents = new ArrayList<>(file.getContent());
+                                    contents.add(idx, comment);
+                                    return file.withContent(contents);
+                                } else {
+                                    throw new RuntimeException("Entry must be present in the properties file!");
+                                }
+                            }
+                        }.visitNonNull(source, ctx);
+                    }
+                } else if (APP_YAML_FILE_REGEX.matcher(fileName).matches() && source instanceof Yaml) {
+                    Set<Yaml.Block> foundEntriesValues = FindProperty.find((Yaml) source, PROPERTY_KEY, false);
+                    if (!foundEntriesValues.isEmpty()) {
+                        projectsWithCommentOverProperty.add(javaProject);
+                        return (SourceFile) new YamlIsoVisitor<ExecutionContext>(){
+                            @Override
+                            public Yaml.Mapping.Entry visitMappingEntry(Yaml.Mapping.Entry entry, ExecutionContext context) {
+                                if (foundEntriesValues.contains(entry.getValue())) {
+//                                    return maybeAutoFormat(entry, entry.withPrefix("\n#" + MIGRATION_MESSAGE + entry.getPrefix()), ctx);
```

Comment:
can you double-check this, the autoformat will ensure the comment has the correct indent

Reply:
If I try the `autoFormat` like below:
```
return maybeAutoFormat(entry, entry.withPrefix("\n#" + PROPS_MIGRATION_MESSAGE + entry.getPrefix()), ctx);
```
I get one of my tests failing which shows that the comment is not indented and the next line indentation is broken as well. Seemed like a bug to me.

Reply:
In this case autoFormat needs to be aware of the parent tree.  Try invoking autoFormat with the parent-cursor
```java
return maybeAutoFormat(entry, entry.withPrefix("\n#" + PROPS_MIGRATION_MESSAGE + entry.getPrefix()),
  ctx, getCursor().getParent());
```

Reply:
Doesn't help, still the comment not indented. Event if drop parent until Yaml (the root). I've added code to get the indent without auto format. Seems like auto format does not work with comments. I quickly looked at format tests in yaml rewrite and did not notice any tests with comments...
I don't think it is a big deal if this comment is not indented :-)

Reply:
Returning as agreed: 
```
return maybeAutoFormat(entry, entry.withPrefix("\n#" + PROPS_MIGRATION_MESSAGE + entry.getPrefix()),
  ctx, getCursor().getParent());
```

Reply:
Hi @BoykoAlex, after giving this a bit more thought, I agree with your original approach of not using autoFormat. Given that the Yaml model only supports comments within the prefix of an element, it's most likely not possible to auto-format comments.